pkg/agent/agent.go

```diff
@@ -434,6 +440,11 @@ func (c *agentController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 
 	if !hc.GetDeletionTimestamp().IsZero() {
 		c.log.Info(fmt.Sprintf("hostedcluster %s has deletionTimestamp %s. Skip reconciling klusterlet secrets", hc.Name, hc.GetDeletionTimestamp().String()))
+
+		if err := c.deleteManagedCluster(ctx, hc); err != nil {
+			c.log.Error(err, "failed to delete the managed cluster")
```

Comment:
@philipwu08 Since this just logs the error and continue, I think it should work on a remote hosting cluster where `ManagedCluster` kind does not exist. It will be nice to ensure this change does not break the addon agent on a remote hosting cluster.